File: Freedom/聚合数据/用户中心/JuheUserViewController.m
//  JuheUserViewController.m
//  Created by Super on 16/9/5.
//  Copyright © 2016年 Super. All rights reserved.
//
#import "JuheUserViewController.h"
#import <XCategory/UILabel+expanded.h>
@interface JuheUserViewCell:BaseTableViewCell
@end
@implementation JuheUserViewCell
-(void)initUI{
    [super initUI];
    self.icon.frame = CGRectMake(10, 10, 40, 40);
    self.title.frame = CGRectMake(XW(self.icon)+10, 20, 200, 20);
    self.line.frame = CGRectMake(10, 59, APPW-20, 1);
}
-(void)setDataWithDict:(NSDictionary *)dict{
    self.icon.image = [UIImage imageNamed:dict[@"pic"]];
    self.title.text = dict[@"name"];
}
@end
@implementation JuheUserViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    [self setTitle:@"个人中心"];
    self.tableView = [[BaseTableView alloc]initWithFrame:CGRectMake(0, 0, APPW, APPH-TabBarH)];
    UIView *headerView = [[UIView alloc]initWithFrame:CGRectMake(0, 0, APPW, 260)];
    UIImageView *icon = [[UIImageView alloc]initWithFrame:CGRectMake(10, 15, 60, 60)];
    icon.layer.cornerRadius = H(icon)/2.0;icon.layer.masksToBounds = YES;
    UILabel *name = [UILabel labelWithFrame:CGRectMake(XW(icon)+10, 10,300, 20) font:fontTitle color:blacktextcolor text:@"用户名：18326891683  👑已认证"];
    UILabel *openid = [UILabel labelWithFrame:CGRectMake(X(name), YH(name), 400, 20) font:fontTitle color:blacktextcolor text:@"OpenId:JH12bd23ef316e3d8a9dfe7402ef8bc453"];
    UILabel *email = [UILabel labelWithFrame:CGRectMake(X(name), YH(openid), 300, 20) font:fontTitle color:blacktextcolor text:@"绑定邮箱:1069106050@qq.com"];
    UILabel *phone = [UILabel labelWithFrame:CGRectMake(X(name), YH(email), 300, 20) font:fontTitle color:blacktextcolor text:@"手机号码:18721064516"];
    [headerView addSubviews:icon,name,openid,email,phone,nil];
    [icon imageWithURL:@"" useProgress:NO useActivity:NO];
    UIView *v = [[UIView alloc]initWithFrame:CGRectMake(0, 0, APPW, 50)];
    UIButton *quit = [[UIButton alloc]initWithFrame:CGRectMake(APPW/2-50, 10, 100, 30)];
    [quit setTitle:@"退      出" forState:UIControlStateNormal];
    [quit setTitleColor:whitecolor forState:UIControlStateNormal];
    quit.backgroundColor = redcolor;[v addSubview:quit];
    NSArray *titles = @[@"我的聚合",@"我的数据",@"我的收藏",@"我的余额",@"聚合币",@"优惠券",@"发票管理",@"其他信息"];
    NSArray *icons = @[@"juheintopublic",@"juheintopublic",@"juheintopublic",@"juheintopublic",@"juheintopublic",@"juheintopublic",@"juheintopublic",@"juheintopublic"];
    BaseScrollView *banItemSV = [BaseScrollView sharedBaseItemWithFrame:CGRectMake(0, 100, APPW, 160) icons:icons titles:titles size:CGSizeMake(APPW/4.0, 80) round:NO];
    [headerView addSubview:banItemSV];
    banItemSV.selectBlock = ^(NSInteger index, NSDictionary *dict) {
        DLog(@"点击了%ld,%@",index,dict);
    };
    self.tableView.dataArray = [NSMutableArray arrayWithObjects:
         @{@"name":@"我的充值记录",@"pic":@"juheintopublic"},@{@"name":@"我的消费记录",@"pic":@"juhelookhistory"},@{@"name":@"账户信息",@"pic":@"juheaboutus"},@{@"name":@"密码修改",@"pic":PuserLogo},
         @{@"name":@"实名认证",@"pic":PuserLogo}, nil];
    [self fillTheTableDataWithHeadV:headerView footV:v canMove:NO canEdit:NO headH:0 footH:0 rowH:60 sectionN:1 rowN:5 cellName:@"JuheUserViewCell"];
    [self.view addSubview:self.tableView];
    self.tableView.delegate = self;
    self.tableView.dataSource = self;
}
-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    DLog(@"点击了第%ld个单元格",indexPath.row);
}
@end

File: Freedom/聚合数据/用户中心/JuheUserViewController.swift
//
//  JuheUserViewController.swift
//  Freedom
import UIKit
import XExtension
import XCarryOn
class JuheUserViewCell:BaseTableViewCell{
    override func initUI() {
        self.icon = UIImageView(frame: CGRect(x: 10, y: 10, width: 40, height: 40))
        self.title = UILabel(frame: CGRect(x:XW(self.icon)+10, y:20, width:200, height: 20))
        self.line = UIView(frame: CGRect(x: 10, y: 59, width: APPW-20, height: 1))
        self.addSubviews([self.title,self.icon,self.line])
        self.title.text = "name"
        self.icon.image = UIImage(named:"taobaomini2")
    }
}
class JuheUserViewController: JuheBaseViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        self.cellHeight = 60
        title = "个人中心"
        self.tableView = BaseTableView(frame: CGRect(x: 0, y: 0, width: APPW, height: APPH-TopHeight))
    let headerView = UIView(frame: CGRect(x: 0, y: 0, width: APPW, height: 260))
    let icon = UIImageView(frame: CGRect(x: 10, y: 15, width: 60, height:60))
    icon.layer.cornerRadius = 30
    icon.layer.masksToBounds = true
        let name = UILabel(frame: CGRect(x: XW(icon)+10, y: 10, width: 300, height: 20))
    name.text = "用户名：18326891683  👑已认证"
        let openid = UILabel(frame: CGRect(x: name.frame.origin.x, y:YH(name), width: 400, height: 20))
        openid.text = "OpenId:JH12bd23ef316e3d8a9dfe7402ef8bc453"
        let email = UILabel(frame: CGRect(x: name.frame.origin.x, y: YH(openid), width: 300, height: 20))
    email.text = "绑定邮箱:1069106050@qq.com"
        let phone = UILabel(frame: CGRect(x: name.frame.origin.x, y: YH(email), width: 300, height: 20))
       phone.text = "手机号码:18721064516"
    headerView.addSubviews([icon,name,openid,email,phone])
    icon.image = UIImage(named:"userLogo")
    let v = UIView(frame: CGRect(x: 0, y: 0, width: APPW, height: 50))
    let quit = UIButton(frame: CGRect(x: APPW/2-50, y: 10, width: 100, height: 30))
        quit.setTitle("", for: .normal)
        quit.setTitleColor(.white, for: .normal)
        quit.backgroundColor = .red
        v.addSubview(quit)
    let titles = ["我的聚合","我的数据","我的收藏","我的余额","聚合币","优惠券","发票管理","其他信息"]
    let icons = ["juheintopublic","juheintopublic","juheintopublic","juheintopublic","juheintopublic","juheintopublic","juheintopublic","juheintopublic"]
    let banItemSV = BaseScrollView(baseItem: CGRect(x: 0, y: 100, width: APPW, height: 160), icons: icons, titles: titles, size: CGSize(width: APPW/4.0, height: 80), round:false)
        headerView.addSubview(banItemSV)
        banItemSV.selectBlock = {(index,dict) in
            print("点击了\(index),\(String(describing: dict))")
        }
        self.tableView.dataArray = NSMutableArray(array: ["我的充值记录","我的消费记录","账户信息","密码修改","实名认证"])
        self.tableView.delegate = self;
        self.tableView.dataSource = self;
        view.addSubview(tableView)
    }
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        var cell = tableView.dequeueReusableCell(withIdentifier: JuheUserViewCell.identifier()) as? JuheUserViewCell
        if cell == nil{
            cell = JuheUserViewCell.getInstance() as? JuheUserViewCell
        }
        return cell!
    }
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        print("点击了第\(indexPath.row)个单元格")
    }
}
